src/components/day.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Cell from './cell';
+
+function Day({ row, mode, changeRow }) {
+  const [range, setRange] = useState({
+    start: null,
+    end: null
+  });
+  const [mouseStatus, setMouseStatus] = useState({
+    down: false
+  });
+
+  function handleMouseClick(cellID) {
+    changeRow((prev) => {
+      return prev.map((cell) => {
+        if (cellID !== cell.id) {
+          return cell;
+        }
+        return {
+          ...cell,
+          mode: mode.mode
+        };
+      });
+    });
+  }
+
+  function handleMouseDown(cell) {
+    setRange({
+      start: { ...cell },
+      end: null
+    });
+    setMouseStatus({ down: true });
+  }
+  function handleMouseMove(cell) {
+    setRange((prev) => ({
+      ...prev,
+      end: { ...cell }
+    }));
+  }
+
+  function setRangeMinToMax(startID, endID) {
+    //function checks which one is bigger than the other, and returns => min, max
+
+    if (startID > endID) {
+      return {
+        start: endID,
+        end: startID
+      };
+    }
+    return {
+      start: startID,
+      end: endID
+    };
+  }
+  useEffect(() => {
+    document.addEventListener('mouseup', () => {
+      setMouseStatus({ down: false });
+      setRange({ end: null, start: null });
+      changeRow((prev) => {
+        return prev.map((cell) => {
+          let newCell = { ...cell };
+          newCell.selected = false;
+          return newCell;
+        });
+      });
+    });
+  }, [changeRow]);
```

Comment:
🚨 Here you have a memory leak. Each time this effect runs, a new event list is created. When your app grows you'll start seeing bugs here - slower app, too many events listeners, etc...

So, in useEffects you must cleanup your effect, in this case, remove the event listener. Learn how to [clean manual event listeners](https://www.pluralsight.com/guides/how-to-cleanup-event-listeners-react). Please do this before merging this MR.